### DISCUSSION TITLE
Feedback about this project

### DISCUSSION THREAD

**Discussion by williamdes** [air-balloon/debian-dashboard#1] [General]
Ref: https://github.com/paultag/libolla/issues/7

**Comment by williamdes** [comment]
Hi @paultag

> (I hope I can manage this without a script?)

Yeah sure :smile:

See: https://github.com/air-balloon/debian-dashboard/blob/main/qa-report.php

Some days ago it was manual, I automated this today.

This is an example of FTP RM report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1028903

I already made a bunch of RM requests: https://bugs.debian.org/cgi-bin/pkgreport.cgi?src=ftp.debian.org;submitter=williamdes%40wdes.fr

As a FTP master member your feedback on such requests I precious !

And I agree with Paul Wise, it should go into Debian QA repos. I will do it when it's a bit more polished up.
Ref: https://lists.debian.org/debian-devel-games/2023/01/msg00010.html

**Comment by paultag** [comment]
(I don't know if this is with or without my ftpteam hat on, so let's default to no ftpteam hat on except when it has to be on - I sent this URL to the team for viz)

@williamdes I don't have a ton of bandwidth for feedback on this tool, but here's a first pass without looking deeply at the source, approach, or even the outputs beyond what you've listed above and the rm bug you linked that I opened:

 - pabs is usually right, you should listen to him :)
 - RoQA is exactly that - removals requested by the QA team. If the QA agrees and executes on the criteria, I don't know that my opinion as to the criteria matters
 - dak will auto-close bugs (`dak rm -D`) associated with the package, no need to include those in the email

The checklist is confusing to my caveman brain. Some checks are good ("`Debian copyright file uses DEP-5`") and others are bad ("`Broken Vcs Target`"). If checks are all checked or all unchecked, it should tell me it's in fairly good shape or totally effed, rather than having to read each item.

About the checklist:
  - Some of the critera is also not clear ("`Not maintained`" - as is, it's got an O bug under WNPP? What about RFA? Is it just the last-uploaded date? The last time the *maintainer* uploaded anything to Debian? - can you link to those bugs? Public MIA ephemera?).
 - It's not clear to me what upstream tests have to do with removal criteria, either.
 - no package in debian is missing a d/copyright - have you seen any? *that* would be a useful bug filing right today. You can set the severity to `serious`, and `X-Debbugs-CC: ftpmaster@ftp-master.debian.org`. Also CC me directly if you want, that's one I'd very much like to get after ASAP.
 - The suites that the package is in don't impact this process. RoM / RoQA removal bugs only impact sid (experimental if the package is there, as was the case with `olla`). We do not remove from `stable` without talking to the stable release team.
 - Some of those checkboxes are optional. `DEP-5`, for instance, is strongly encouraged but not mandatory (yet?). Are they indicators or removal criteria?

I'd say that on the whole this is not disruptive really - and I think perhaps net good if the QA team is onboard with critera and the MBF is fully coordinated -- but there's also a *lot* of noise in the removal bug. I just mostly found it funny that I was getting a request to a file a bug for me that ... I would have to deal with :)

On the whole, I'll leave you with this -- the removal bug is more of a "tombstone" that everything points to as a description of why the removal action was taken, and why the removal was the thing we came to, rather than letting it continue to exist or maintain it. I say focus on cleaning up the bug text a bit, try to make it easier to do a quick scan of and walk away understanding why specifically the removal was required.

**Reply by williamdes** [reply]
> * pabs is usually right, you should listen to him :)

👍🏻 

> * RoQA is exactly that - removals requested by the QA team. If the QA agrees and executes on the criteria, I don't know that my opinion as to the criteria matters

About that Scott Kitterman told me to use RoQA instead of RoDM (Request of Debian Maintainer).
So I am for now an alone person acting as/for QA.

/cc @kitterma if you want to comment on this thread.

> * dak will auto-close bugs (`dak rm -D`) associated with the package, no need to include those in the email

Thanks, I will provide a list of bugs only when they are relevant to the request.
I was wondering since the start if this was useful.

> The checklist is confusing to my caveman brain. Some checks are good ("`Debian copyright file uses DEP-5`") and others are bad ("`Broken Vcs Target`"). If checks are all checked or all unchecked, it should tell me it's in fairly good shape or totally effed, rather than having to read each item.

I agree, I also struggle with my own list. I need to refactor it. Maybe adding `PASS` or `FAIL` at the end or start.

> About the checklist:
> 
> * Some of the critera is also not clear ("`Not maintained`" - as is, it's got an O bug under WNPP? What about RFA? Is it just the last-uploaded date? The last time the _maintainer_ uploaded anything to Debian? - can you link to those bugs? Public MIA ephemera?).

Good point, I will work on this

> * It's not clear to me what upstream tests have to do with removal criteria, either.

It makes sense (only?) for me, because if upstream tests are in the package but not used that's a bit sad.
It means the package **could** has autopkgtests.

> * no package in debian is missing a d/copyright - have you seen any? _that_ would be a useful bug filing right today. You can set the severity to `serious`, and `X-Debbugs-CC: ftpmaster@ftp-master.debian.org`. Also CC me directly if you want, that's one I'd very much like to get after ASAP.

Not for now, it just prevents a silent script failure for now. That said some copyright files are a joke, they document not much.
Like this one: https://sources.debian.org/src/printfilters-ppd/2.13-11.1/debian/copyright/

> * The suites that the package is in don't impact this process. RoM / RoQA removal bugs only impact sid (experimental if the package is there, as was the case with `olla`). We do not remove from `stable` without talking to the stable release team.

Sure, for me it indicates that the package "was" in Debian or not at all. Some packages in unstable never migrating for example.

> * Some of those checkboxes are optional. `DEP-5`, for instance, is strongly encouraged but not mandatory (yet?). Are they indicators or removal criteria?

They are quality indicators. My first goal at this project was to have as much indicators and then narrow it down to file FTP RM requests.

> I'd say that on the whole this is not disruptive really - and I think perhaps net good if the QA team is onboard with critera and the MBF is fully coordinated -- but there's also a _lot_ of noise in the removal bug. I just mostly found it funny that I was getting a request to a file a bug for me that ... I would have to deal with :)

The first part of your sentence is unclear to me :/

> On the whole, I'll leave you with this -- the removal bug is more of a "tombstone" that everything points to as a description of why the removal action was taken, and why the removal was the thing we came to, rather than letting it continue to exist or maintain it. I say focus on cleaning up the bug text a bit, try to make it easier to do a quick scan of and walk away understanding why specifically the removal was required.

" I say focus on cleaning up the bug text a bit, try to make it easier to do a quick scan of and walk away understanding why specifically the removal was required.".
Yes, this is the goal 🚀

**Comment by williamdes** [comment]
I moved the project to https://salsa.debian.org/williamdes/debian-dashboard
